recipes/hopsmon.rb

```diff
@@ -0,0 +1,62 @@
+#Generate and sign certificates for hopsmon
+hopsworks_ip = private_recipe_ip('hopsworks', 'default')
+hopsworks_https_port = 8181
+if node.attribute?('hopsworks')
+  if node['hopsworks'].attribute?('https') and node['hopsworks']['https'].attribute?('port')
+    hopsworks_https_port = node['hopsworks']['https']['port']
+  end
+end
+
+node.override['kube-hops']['pki']['ca_api'] = "#{hopsworks_ip}:#{hopsworks_https_port}"
+
+hopsmon_kube_certs_dir = "#{node['kube-hops']['monitoring']['certs-dir']}"
+directory hopsmon_kube_certs_dir do
+  owner node['kube-hops']['user']
+  group node['kube-hops']['user']
+  mode '0700'
+  action :create
+  not_if { ::File.directory?(node['kube-hops']['monitoring']['certs-dir']) }
+end
+
+kube_hops_certs 'hopsmon' do
+  path        hopsmon_kube_certs_dir
+  subject     "/CN=hopsmon"
+  not_if      { ::File.exist?("#{node['kube-hops']['monitoring']['cert-crt']}") }
+end
+
+bash 'hopsmon-kube-certs-dirs-permissions' do
+  user "root"
+  group "root"
+  code <<-EOH
+    setfacl -m u:#{node['kube-hops']['monitoring']['user']}:rx #{node['kube-hops']['dir']}
+    setfacl -m u:#{node['kube-hops']['monitoring']['user']}:rx #{node['kube-hops']['hops-system']['base_dir']}
+    setfacl -m u:#{node['kube-hops']['monitoring']['user']}:rx #{node['kube-hops']['monitoring']['certs-dir']}
+    setfacl -m u:#{node['kube-hops']['monitoring']['user']}:rx #{node['kube-hops']['monitoring']['cert-crt']}
+    setfacl -m u:#{node['kube-hops']['monitoring']['user']}:rx #{node['kube-hops']['monitoring']['cert-key']}
+  EOH
+end
```

Comment:
This PR has huge side effect that it only works as long as Prometheus and K8s master are colocated. It's definitely not great, but I can't think of a solution right now that would work.

That being said. Giving the permissions to the hopsmon user should be done on the hopsmon recipe, rather than here I believe.
Now we have 2 hopsmon users, one linux user, which is actually defined in hopsmonitor-chef and the K8s one. Which doesn't have to match. If by any chance you need to change it, good luck.

If you move the permissions changes to hopsmonitor-chef and make it dependent on kube-hops-chef, this would not be a problem.